src/conduits/RwaInputConduit3.t.sol

```diff
@@ -256,22 +258,101 @@ contract RwaInputConduit3Test is Test, DSMath {
         assertEq(testUrn.balance(address(dai)), 500 ether);
     }
 
-    function testRevertOnSwapAboveLine() public {
+    function testPushAmountWhenHaveSomeDaiBalanceGetExectAmount() public {
+        dai.mint(address(inputConduit), 100 ether);
+        assertEq(dai.balanceOf(address(inputConduit)), 100 ether);
+
         assertEq(usdx.balanceOf(me), USDX_MINT_AMOUNT);
         assertEq(usdx.balanceOf(address(inputConduit)), 0);
         assertEq(usdx.balanceOf(address(joinA)), 0);
 
-        usdx.transfer(address(inputConduit), USDX_MINT_AMOUNT);
+        usdx.transfer(address(inputConduit), 500 * USDX_BASE_UNIT);
+
+        assertEq(usdx.balanceOf(me), USDX_MINT_AMOUNT - 500 * USDX_BASE_UNIT);
+        assertEq(usdx.balanceOf(address(inputConduit)), 500 * USDX_BASE_UNIT);
+
+        assertEq(testUrn.balance(address(dai)), 0);
+
+        vm.expectEmit(true, true, false, false);
+        emit Push(address(testUrn), 500 ether);
+        inputConduit.push(500 * USDX_BASE_UNIT);
+
+        assertEq(usdx.balanceOf(address(joinA)), 500 * USDX_BASE_UNIT);
+        assertEq(usdx.balanceOf(address(inputConduit)), 0);
```

Comment:
not critical but do we want to use PSM_LINE instead of ether in the test assertions

Reply:
Hmm... not sure here it's make some difference

Reply:
We sue it there: https://github.com/clio-finance/mip21-toolkit/pull/13/files/35e4331a2d8d4f93b6114e8cd10cc8a6b8768eea#diff-8c1a0f9fee9305f5ba8211ce53ae691bab248b011fd769488d0b17e4cc3a7cb6R336